testutil/keeper/supplier.go

```diff
@@ -38,6 +50,40 @@ func SupplierKeeper(t testing.TB) (*keeper.Keeper, sdk.Context) {
 	mockBankKeeper.EXPECT().DelegateCoinsFromAccountToModule(gomock.Any(), gomock.Any(), types.ModuleName, gomock.Any()).AnyTimes()
 	mockBankKeeper.EXPECT().UndelegateCoinsFromModuleToAccount(gomock.Any(), types.ModuleName, gomock.Any(), gomock.Any()).AnyTimes()
 
+	mockSessionKeeper := mocks.NewMockSessionKeeper(ctrl)
+	mockSessionKeeper.EXPECT().
+		GetSession(gomock.AssignableToTypeOf(sdk.Context{}), gomock.Any()).
+		DoAndReturn(
+			func(
+				ctx sdk.Context,
+				req *sessiontypes.QueryGetSessionRequest,
+			) (*sessiontypes.QueryGetSessionResponse, error) {
+				sessionMock, ok := sessionFixtures[req.GetApplicationAddress()]
+				require.Truef(t, ok, "application address not provided during mock construction: %q", req.ApplicationAddress)
+
+				return &sessiontypes.QueryGetSessionResponse{
+					Session: &sessiontypes.Session{
+						Header: &sessiontypes.SessionHeader{
+							ApplicationAddress:      sessionMock.AppAddr,
+							Service:                 req.GetService(),
+							SessionStartBlockHeight: 1,
+							SessionId:               sessionMock.SessionId,
+							SessionEndBlockHeight:   5,
+						},
+						SessionId:           sessionMock.SessionId,
+						SessionNumber:       1,
+						NumBlocksPerSession: 4,
```

Comment:
Where would these integers be better sourced from?

Reply:
Sorry, I'm not following. Can you elaborate?

Reply:
I've refactored to not need a new type. It seems like that was mostly what you were getting at.